### ISSUE TITLE
Uploading large objects, e.g. 3GB, is very slow compared to Python client? 

### ISSUE THREAD

**Issue by ssami** [javaswift/joss#124]
I'm not sure if this is technically an issue, but as I don't see any documentation that helps me with this (or forums?) I hope someone can correct my understanding. 

I'm trying to find out how I can upload and download objects that are large (3++ GB) efficiently. I understand that I can segment the object and according to the Swift documentation, upload them in parallel.  http://docs.openstack.org/developer/swift/overview_large_objects.html says:
`This would split the large_file into 1G segments and begin uploading those segments in parallel.`

So I took my 3.1GB large file and split it into segments of 1MB (though that might be overkill). Then I used UploadInstructions to set that 1MB segment size, then called uploadObject. 

```
Container test = account.getContainer("test");
StoredObject newObject = test.getObject("big-file");
// Upload instructions to split file and upload in many segments
UploadInstructions instructions = new UploadInstructions(new File("mt-model.tar.gz"));
instructions.setSegmentationSize(1000000L);
SegmentationPlan plan = instructions.getSegmentationPlan();
 
newObject.uploadObject(instructions);
```

But this line in AbstractContainer https://github.com/javaswift/joss/blob/master/src/main/java/org/javaswift/joss/client/core/AbstractContainer.java#L180 seems to be doing sequential uploads of every segment. 

Is my understanding correct? If yes, doesn't this make Joss much slower than the Python client?
If not, what can I fix in my code that enables me to efficiently upload large objects?

**Comment by ferrys** [comment]
Hi @ssami, your understanding is correct and this is how JOSS works. It could be a great enhancement to speed this up using the Python client as an example. Would this be something you would be interested in contributing?

**Comment by jelmerk** [comment]
Is this still not implemented ? Having this exposed in the api and then not doing the work in parallel seems kind of pointless

**Comment by anil2104012** [comment]
Any plans to fix this issue ?
